test/unit/utils-specs.js

```diff
@@ -72,4 +77,76 @@ describe('util', () => {
       execStub.threw().should.be.true;
     });
   });
+
+  describe('simExists', () => {
+    const devicesStub = {
+    '7.1': [
+     { name: 'iPhone 5',
+       udid: 'B236B73C-8EFA-4284-AC1F-2A45F3286E4C',
+       state: 'Shutdown' },
+     { name: 'iPhone 5s',
+       udid: '8E248C90-0F79-46AD-9CAA-8DF3B6E3FBA6',
+       state: 'Shutdown' },
+     { name: 'iPad Air',
+       udid: 'FA5C971D-4E05-4AA3-B48B-C9619C7453BE',
+       state: 'Shutdown' } ],
+    '8.1': [
+     { name: 'iPhone 5',
+       udid: 'B5048708-566E-45D5-9885-C878EF7D6D13',
+       state: 'Shutdown' },
+     { name: 'iPhone 5s',
+       udid: '2F7678F2-FD52-497F-9383-41D3BB401FBD',
+       state: 'Shutdown' },
+     { name: 'iPhone 6 Plus',
+       udid: '013D6994-B4E6-4548-AD77-C0D7C6C6D245',
+       state: 'Shutdown' } ],
+    '8.3': [
+     { name: 'iPhone 5',
+       udid: '813AAB6A-32C8-4859-A5CF-F3355C244F54',
+       state: 'Shutdown' },
+     { name: 'iPhone 5s',
+       udid: '9D3A405E-65D6-4743-85DA-E644DA9A8373',
+       state: 'Shutdown' },
+     { name: 'iPhone 6 Plus',
+       udid: 'D94E4CD7-D412-4198-BCD4-26799672975E',
+       state: 'Shutdown' },
+     { name: 'iPhone 6',
+       udid: '26EAADAE-1CD5-42F9-9A4C-50554CDF0910',
+       state: 'Shutdown' },
+     { name: 'iPad 2',
+       udid: 'C8E68217-82E6-42A8-8326-9824CA2E7C7C',
+       state: 'Shutdown' } ]
+  };
+
+  it('returns true if device is found', async () => {
+    getDevicesStub.returns(Promise.resolve(devicesStub));
+    let existence = [];
+     existence.push(simExists('D94E4CD7-D412-4198-BCD4-26799672975E'));
+     existence.push(simExists('C8E68217-82E6-42A8-8326-9824CA2E7C7C'));
+     existence.push(simExists('B5048708-566E-45D5-9885-C878EF7D6D13'));
+     existence.push(simExists('8E248C90-0F79-46AD-9CAA-8DF3B6E3FBA6'));
+
+     let results = await B.all(existence);
+
+     for (let result of results) {
+       result.should.be.true;
+     }
+  });
+
+  it('returns false if device is not found', async () => {
+    getDevicesStub.returns(Promise.resolve(devicesStub));
+    let existence = [];
```

Comment:
why not just initialize all these when the array is initialized? It's not dynamic at all.